assets/src/scripts/monitoring-location/components/hydrograph/date-controls.js

```diff
@@ -31,32 +34,177 @@ export const drawDateRangeControls = function(elem, store, siteno) {
+        .text(`${MAX_NUMBER_OF_DAYS_USERS_CAN_INPUT} digits allowed`)
+        .attr('id', 'with-hint-input-days-from-today-info')
+        .attr('class', 'usa-hint usa-character-count__message')
+        .attr('aria-live', 'polite');
+    // Create a validation alert for user selection of number of days before today
+    const customDaysBeforeTodayValidationContainer = containerCustomDaysBeforeToday.append('div')
+        .attr('class', 'usa-alert usa-alert--warning usa-alert--validation')
+        .attr('id', 'custom-days-before-today-alert-container')
+        .attr('hidden', true);
+    const customDaysBeforeTodayAlertBody = customDaysBeforeTodayValidationContainer.append('div')
+        .attr('class', 'usa-alert__body')
+        .attr('id', 'custom-days-before-today-alert');
+    customDaysBeforeTodayAlertBody.append('h3')
+        .attr('class', 'usa-alert__heading')
+        .text('Requirements');
+    numberOfDaysSelection.call(link(store, (container, {userInputTimeRangeSelectionButton, userInputCustomTimeRangeSelectionButton, userInputNumberOfDays}) => {
+        if (userInputTimeRangeSelectionButton === 'custom' && userInputCustomTimeRangeSelectionButton === 'days-input') {
+                container.select('#with-hint-input-days-from-today')
+                    .property('value', userInputNumberOfDays);
+        }
+    }, createStructuredSelector({
+        userInputTimeRangeSelectionButton: getUserInputTimeRangeSelectionButton,
+        userInputCustomTimeRangeSelectionButton: getUserInputCustomTimeRangeSelectionButton,
+        userInputNumberOfDays: getUserInputNumberOfDays
+    })));
+
+    // Adds controls for the 'days before today' submit button
+    const daysBeforeTodaySubmitContainer = containerCustomDaysBeforeToday.append('div')
+        .attr('class', 'submit-button');
+    daysBeforeTodaySubmitContainer.append('button')
+        .attr('class', 'usa-button')
+        .attr('id', 'custom-date-submit-days')
+        .text('Display data on graph')
+        .on('click', function() {
+            const userSpecifiedNumberOfDays = document.getElementById('with-hint-input-days-from-today').value;
+            const formattedPeriodQueryParameter = `P${parseInt(userSpecifiedNumberOfDays)}D`;
+            // Validate user input for things not a number and blank entries
+            if (isNaN(userSpecifiedNumberOfDays) || userSpecifiedNumberOfDays.length === 0) {
+                customDaysBeforeTodayAlertBody.selectAll('p').remove();
+                customDaysBeforeTodayAlertBody.append('p')
+                    .text('Entry must be a number.');
+                customDaysBeforeTodayValidationContainer.attr('hidden', null);
+            } else {
+                customDaysBeforeTodayValidationContainer.attr('hidden', true);
+                store.dispatch(ivTimeSeriesStateActions.setUserInputNumberOfDays(userSpecifiedNumberOfDays));
+                store.dispatch(ivTimeSeriesDataActions.retrieveExtendedIVTimeSeries(
```

Comment:
I don't think you should be using retrieveExtendedIVTimeSeries. This will also retrieve the compare data (one year in the future) and we are not allowing that for custom time period or date ranges. I think you should be using retrieveCustomTimePeriodIVTimeSeries.

Reply:
Finally got this working. Thanks for the help!